src/Utils.vala

```diff
@@ -60,12 +60,15 @@ namespace Utils {
         if (Installer.App.test_mode) {
             critical (_("Test mode switch user"));
         } else {
-            get_system_instance ();
+            // This touches the file `/var/lib/lightdm/demo-mode`, which signals to the greeter that the next session it launches
+            // should be the live (demo) session. If this file doesn't exist, it just relaunches the installer session
+            Installer.Daemon.get_default ().trigger_demo_mode.begin ();
```

Comment:
Should we make sure that this got correctly executed (and that we have no racing) before doing anything and log the error with `debug`